#   Tarea numero 2
### Serie de Taylor
### Jose David Ramos Rios
### Codigo: 161002940

### Tabla -------------------------------------------------------------------------------------------------------------------
Una Serie de Taylor es la representación de una función como la sumatoria infinita de términos, calculados a partir de las derivadas de la función original evaluada en un punto.

$$
\begin{eqnarray*}
f(x) &=& f(a)+\frac {f'(a)}{1!} (x-a)+ \frac{f''(a)}{2!} (x-a)^2+\frac{f'''(a)}{3!}(x-a)^3+ \cdots \\
     &=& \sum_{n=0} ^ {\infty} \frac {f^{(n)}(a)}{n!} \, (x-a)^{n} 
\end{eqnarray*}
$$


# para el desarrollo de serie de taylor se obtiene que 

\begin{multline*}
cos(x) = 1 -\frac{x^2}{2!} + \frac{x^4}{4!}-\frac{x^6}{6!}+\frac{x^8}{8!}
\end{multline*}

esto debido a que la derivada impares de coseno al ser evaludas en cero su valor va a ser cero 
y lo que vamos a obtener son en la parte par de la devidas en este caso un valor oscilatorio 1,-1


In [44]:
# serie de taylor 
num = float(input("ingrese valor x: "))
ap= float(input("valor aproximado: "))


if(ap==1):
    a=1
    print(a)

if(ap==2):
    a=1+0
    print(a)
    
if(ap==3):
    a= 1+ 0 -((num**2)/2)
    print(a)    

if(ap==4):
    a= 1+ 0 -((num**2)/2)+0
    print(a)  

if(ap==5):
    a= 1+ 0 -((num**2)/2)+0+((num**4)/24)
    print(a)

ingrese valor x: 10
valor aproximado: 5
367.666666667


# serie de Taylor
Ejercicio:
 
Complete la siguiente tabla para la función $cos(x)$, alrededor del punto $a = 0$, utilizando el número indicado de términos en la aproximación.

 
   | x     | Exacto |Aprox.1|Aprox.2|Aprox.3|Aprox.4|Aprox.5|
   |:----------------:|:--------------:|:-----------:|:------------:|:----------:|:-----:|:--------|
   |*0 *|1 |1|1|1|1|1|
   |*0.00001*|1|1|1|0.99999999995|0.99999999995|0.99999999995|
   |*0.0001*|0.999999995|1|1|0.999999995|0.999999995|0.999999995|
   |*0.001*|0.9999995|1|1|0.9999995|0.9999995|0.9999995|
   |*0.01*|0.9999500004|1|1|0.99995|0.99995|0.99995|
   |*0.1*|0.9950041653|1|1|0.995|0.995|0.995004166667|
   |*1*|0.5403023059|1|1|0.5|0.5|0.541666666667|
   |*10*|-0.8390715291|1|1|-49.0|-49.0|367.666666667|
   Tabla 2
